platform/base.go

```diff
@@ -115,6 +117,29 @@ func (bc *BaseCluster) SSH(m Machine, cmd string) ([]byte, error) {
 	return out, err
 }
 
+// NewSSH is a temporary method as part of refactoring SSH
+func (bc *BaseCluster) NewSSH(m Machine, cmd string) ([]byte, []byte, error) {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	client, err := bc.SSHClient(m.IP())
+	if err != nil {
+		return nil, nil, err
+	}
+	defer client.Close()
+
+	session, err := client.NewSession()
+	if err != nil {
+		return nil, nil, err
+	}
+	defer session.Close()
+
+	session.Stdout = &stdout
+	session.Stderr = &stderr
+	err = session.Run(cmd)
+	out := bytes.TrimSpace(stdout.Bytes())
+	return out, stderr.Bytes(), err
```

Comment:
Maybe I should leave a comment on this one. I'm trimming `stdout` for backwards compatibility.
I think it really should be the caller's responsibility in the first place, so I'm not really happy about the trimming.. hence only doing it where backwards compatibility is a concern.

I could be swayed to trimming both.

Reply:
As it is, `TestCluster.SSH()` will log an extra blank line.

I'd argue the inconsistency is worse than the trimming, and it's convenient to trim in one place rather than all over the codebase.  Do we expect to receive binary data over stdout/stderr?

Reply:
I think it would make sense to either not trim either (and update the PR to Trim the stdout during the refactor) or trim both. Having it be different could be confusing later on.

Reply:
Added a commit to trim both